### PR TITLE
wasmtime: Implement `global.{get,set}` for externref globals

### PR DESCRIPTION
**Pull request by fitzgen**
We use libcalls to implement these -- unlike `table.{get,set}`, for which we
create inline JIT fast paths -- because no known toolchain actually uses
externref globals.

Part of #929

### CODE REVIEW

File: tests/all/gc.rs [hunk 1]
```diff
@@ -517,3 +517,95 @@ fn pass_externref_into_wasm_during_destructor_in_gc() -> anyhow::Result<()> {
...
+        set_g.call(&args)?;
+    }
+
+    // Remove the `GcOnDrop` from the `VMExternRefActivationsTable`.
+    store.gc();
+
+    // Overwrite the `GcOnDrop` global value, causing it to be dropped, and
+    // triggering a GC.
+    set_g.call(&[Val::ExternRef(None)])?;
+    assert_eq!(gc_count.get(), 1);
```
Comment (alexcrichton) [inline tests/all/gc.rs:555]: Could this also assert that `gc_count.get()` is 0 before the call?

File: tests/all/gc.rs [hunk 2]
```diff
@@ -517,3 +517,95 @@ fn pass_externref_into_wasm_during_destructor_in_gc() -> anyhow::Result<()> {
...
+            touched: touched.clone(),
+        })))];
+        set_g.call(&args)?;
+    }
+
+    // Remove the `TouchGlobalOnDrop` from the `VMExternRefActivationsTable`.
+    store.gc();
+
+    set_g.call(&[Val::ExternRef(Some(ExternRef::new("hello".to_string())))])?;
+    assert!(touched.get());
```
Comment (alexcrichton) [inline tests/all/gc.rs:592]: Like above, could this assert that it's not touched before the call?

**Comment by fitzgen** [comment]
My thinking was that, because no toolchains emit `externref` globals, it wasn't worth the effort to make the inline JIT fast path generation code more generic. This would also require some `cranelift-wasm` interface changes.

Do you think it is worth the effort?

**Comment by alexcrichton** [comment]
Nah I don't think it's worth it if it requires larger changes. I just thought that it might be simple to implement today if the code was extracted from table.{get,set}, but if that's not the case then this seems fine to me!

**Comment by fitzgen** [comment]
@alexcrichton good call on adding tests for the global API with funcref and externref, that actually required a bit of plumbing in the trampoline module. Do you want to take a look at the new commit? Thanks!

**Comment by alexcrichton** [comment]
👍

**Pull request merged**